lib/Utils/dir_list_ci.cpp

```diff
@@ -55,59 +103,144 @@ std::string DirListCI::resolveFileCase(const std::string &in_name)
     auto subDir = name.find('/');
     if(subDir != std::string::npos)
     {
+        // THIS BEHAVIOR IS EXPENSIVE AND WASTEFUL, AVOID INVOKING AT ALL COSTS!!!
         auto sdName = resolveDirCase(name.substr(0, subDir));
         DirListCI sd(m_curDir + "/" + sdName);
-        return sdName + "/" + sd.resolveFileCase(name.substr(subDir + 1));
+        std::string found = sd.resolveFileCase(name.substr(subDir + 1));
+        if(found.empty())
+            return "";
+        else
+            return sdName + "/" + found;
     }
 
     // keep MixerX path arguments untouched
     auto pathArgs = name.find('|');
     if(pathArgs != std::string::npos)
     {
         auto n = name.substr(0, pathArgs);
-        for(std::string &c : m_fileList)
-        {
-            if(SDL_strcasecmp(c.c_str(), n.c_str()) == 0)
-                return c + name.substr(pathArgs);
-        }
+        std::string uppercase_string;
+        uppercase_string.resize(n.length());
+        std::transform(n.begin(), n.end(), uppercase_string.begin(),
+            [](unsigned char c){ return std::toupper(c); });
+        auto found = m_fileMap.find(uppercase_string);
+        if(found != m_fileMap.end())
+            return found->second + name.substr(pathArgs);
     }
     else
-    for(std::string &c : m_fileList)
     {
-        if(SDL_strcasecmp(c.c_str(), name.c_str()) == 0)
-            return c;
+        std::string uppercase_string;
+        uppercase_string.resize(name.length());
+        std::transform(name.begin(), name.end(), uppercase_string.begin(),
+            [](unsigned char c){ return std::toupper(c); });
+        auto found = m_fileMap.find(uppercase_string);
+        if(found != m_fileMap.end())
+            return found->second;
     }
 
-    return name;
+    return "";
+}
+
+std::string DirListCI::resolveFileCase(const std::string &in_name)
+{
+#if 0
+    return in_name; // no need on Windows
+#else
```

Comment:
It's time to remove these macros ad this thing is needed for Windows too, mainly because of filenames caching which does work out of FS.

Reply:
Done.